### PR TITLE
Add gulp task dist-api

### PR DESCRIPTION
**Pull request by jon-a-nygaard**
# Description
- [x] Provide a static file server for easy serve of the API
- [x] Zip offline API and add zip to distribution archives
- [x] Add task as a part of the gulp task named `dist`

# Example(s)
Run `npx gulp dist-api` locally too see result in `build/dist/{product}`

# Related issue(s)
- Touch #8533

### CODE REVIEW

File: tools/gulptasks/lib/api/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# How to view the API
+The Highcharts API requires a static files server to be able to perform optimal.
```
Comment (TorsteinHonsi) [inline tools/gulptasks/lib/api/README.md]: _file_ server (singular)

File: tools/gulptasks/lib/api/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# How to view the API
+The Highcharts API requires a static files server to be able to perform optimal. 
+You can use your own static file server if you have one available, but we have also provided a simple Node.js server to get you started quickly.
+
+## Install Node.js
+The provided server requires Node.js to be installed, which can be done by downloading and executing their installer provided at the [Node.js Website](https://nodejs.org/en/)
```
Comment (TorsteinHonsi) [inline tools/gulptasks/lib/api/README.md]: Missing dot at the end of the sentence.

File: tools/gulptasks/lib/api/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+## Start the server
+Once Node.js is installed you can start the server by opening a terminal in this folder, and execute the command `node server.js`. After this the API should be available at
```
Comment (TorsteinHonsi) [inline tools/gulptasks/lib/api/README.md]: ... available at [what?]

File: tools/gulptasks/lib/api/server.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+);
+const MIME_TYPE = {
+    '.css': 'text/css',
+    '.html': 'text/html',
+    '.js': 'text/javascript',
+    '.json': 'application/json',
+    '.png': 'image/png',
+    './svg': 'image/svg+xml'
```
Comment (TorsteinHonsi) [inline tools/gulptasks/lib/api/server.js]: This one doesn't seem to be working, or at least the logo `highcharts.svg` is not working properly. It seems like the server is looking for `highcharts.svg.html` on the file system.

**Comment by jon-a-nygaard** [comment]
Thanks @TorsteinHonsi, please see latest changes.

**Pull request merged**